sdk/storage/Azure.Storage.Blobs/perf/Azure.Storage.Blobs.Perf/Options/PartitionedTransferOptions.cs

```diff
@@ -98,5 +116,203 @@ private void UpdateStorageTransferOptions()
+                    {
+                        var stream = await response.Content.ReadAsStreamAsync();
+                        content = await BinaryData.FromStreamAsync(stream).ConfigureAwait(false);
+                    }
+
+                    return new ResponseSource(response, content);
+                }
+#if NET5_0_OR_GREATER
+                public static ResponseSource Create(HttpResponseMessage response)
+                {
+                    var content = new BinaryData(string.Empty);
+                    if (response.Content != null)
+                    {
+                        var stream = response.Content.ReadAsStream();
+                        content = BinaryData.FromStream(stream);
+                    }
+
+                    return new ResponseSource(response, content);
+                }
+#endif
+            }
+        }
+
+        private class HttpRequestMessageComparer : IEqualityComparer<HttpRequestMessage>
+        {
+            public bool Equals(HttpRequestMessage x, HttpRequestMessage y)
+            {
+                if (ReferenceEquals(x, y))
+                {
+                    return true;
+                }
+
+                if (ReferenceEquals(x, null))
+                {
+                    return false;
+                }
+
+                if (ReferenceEquals(y, null))
+                {
+                    return false;
+                }
+
+                if (x.GetType() != y.GetType())
+                {
+                    return false;
+                }
+
+                return x.Method.Equals(y.Method)
+                    && Equals(x.RequestUri, y.RequestUri)
+                    && x.Version.Equals(y.Version);
+/*
+#if NET5_0_OR_GREATER
+                    && x.VersionPolicy == y.VersionPolicy
+                    && EqualOptions(x.Options, y.Options)
+#else
+                    && EqualOptions(x.Properties, y.Properties)
+#endif
+                    && EqualHeaders(x.Headers, y.Headers)
+                    && EqualContent(x.Content, y.Content);
+*/
+            }
+
+            public int GetHashCode(HttpRequestMessage obj)
+            {
+                var hc1 = (uint)(obj.Method?.GetHashCode() ?? 0);
+                var hc2 = (uint)(obj.RequestUri?.GetHashCode() ?? 0);
+                var hc3 = (uint)(obj.Version?.GetHashCode() ?? 0);
+
+                var hc12 = (hc1 + hc2) * (hc1 + hc2 + 1) / 2 + hc2;
+                var hc123 = (hc12 + hc3) * (hc12 + hc3 + 1) / 2 + hc3;
+
+                return (int)hc123;
+            }
+
+            private static bool EqualOptions(IDictionary<string, object> xOptions, IDictionary<string, object> yOptions)
+            {
+                if (xOptions.Count != yOptions.Count)
+                {
+                    return false;
+                }
+
+                foreach (var kvp in xOptions)
+                {
+                    if (!xOptions.TryGetValue(kvp.Key, out var yValue) || !Equals(kvp.Value, yValue))
```

Comment:
Looks like we are comparing xOptions to xOptions here
```suggestion
                    if (!yOptions.TryGetValue(kvp.Key, out var yValue) || !Equals(kvp.Value, yValue))
```

Reply:
Changes in this file are reverted